tests/tests_update.yml

```diff
@@ -1,20 +1,71 @@
 # SPDX-License-Identifier: MIT
 ---
-- name: Ensure that we can set the policy using this role without reboot
+- name: Ensure that we can set the policy using this role (without reboot)
   hosts: all
-  roles:
-    - role: linux-system-roles.crypto_policies
+  tasks:
+
+    - name: Set correct base policy
+      include_role:
+        name: linux-system-roles.crypto_policies
       vars:
         crypto_policies_policy: LEGACY
         crypto_policies_reload: false
+    - name: Verify that base policy was updated
+      assert:
+        that:
+          - crypto_policies_active == 'LEGACY'
+          - crypto_policies_reboot_required == true
 
-  tasks:
-    - name: Verify the facts are correctly set
+    - name: Set correct base policy and policy module
+      include_role:
+        name: linux-system-roles.crypto_policies
+      vars:
+        crypto_policies_policy: DEFAULT:NO-SHA1
+        crypto_policies_reload: false
+    - name: Verify that base policy and policy module were updated
+      assert:
+        that:
+          - crypto_policies_active == 'DEFAULT:NO-SHA1'
+          - crypto_policies_reboot_required == true
+
+    - name: Setting incorrect base policy should fail
       block:
-        - meta: flush_handlers
+        - name: Set incorrect base policy
+          include_role:
+            name: linux-system-roles.crypto_policies
+          vars:
+            crypto_policies_policy: NOEXIST
+            crypto_policies_reload: false
+          register: role_result
```

Comment:
The `include_role` does not register any summary result like this. I think what is registered is result of the last task in the role, or something as unexpected as this. It is better to rely on the other variables (as you already do) so please, remove this variable.

Reply:
The same for the other block below.